packages/aws-cdk-lib/aws-ec2/lib/security-group.ts

```diff
@@ -369,22 +369,22 @@ export class SecurityGroup extends SecurityGroupBase {
    *
    * @deprecated Use `fromLookupById()` instead
    */
-  public static fromLookup(scope: Construct, id: string, securityGroupId: string) {
-    return this.fromLookupAttributes(scope, id, { securityGroupId });
+  public static fromLookup(scope: Construct, id: string, securityGroupId: string, additionalCacheKey?: string) {
```

Comment:
I'd prefer not to have optional parameters as positionals, because this ultimately leads to `callFunction(3, undefined, undefined, true, undefined, false)` 😉.

If you want to add it here, create an `Options` struct for it. But more pragmatically: it's already available in the `fromLookupAttributes` method, which is good enough. No need to add it here as well then.

Reply:
`fromLookupAttributes` is private (presumably these methods are to differentiate between mutually exclusive valid sets of options), so this is still needed. I'll swap to an options struct